torch_geometric/datasets/rel_link_pred_dataset.py

```diff
@@ -85,7 +85,15 @@ def process(self):
                 kwargs[f'{split}_edge_index'] = torch.tensor([src, dst])
                 kwargs[f'{split}_edge_type'] = torch.tensor(rel)
 
-        data = Data(num_nodes=len(entities_dict), **kwargs)
+        # For message passing, we add reverse edges and types to the graph:
```

Comment:
If this is done in the dataset before the sampling, the negative samples have no reverse edges and are easy to recognize. This should be applied to the negative samples as well. Furthermore, it could be more memory efficient if this is done only on the current batch.

Reply:
Not sure I understand that. While message passing makes use of reverse edges, the decoder does not.

Reply:
Ok, I didn't see that the decoder doesn't get the reverse edges. In my implementation, both (encoder & decoder) got the reverse edges, as in most implementations I saw. The original paper does not look at this in detail. I don't know if we can achieve the same results this way.

Reply:
I think we only need to decode "one direction" as the reverse direction is given by it as well. IMO, reverse edges are only useful to enhance the message passing flow, and shouldn't play any role in the prediction part.

Reply:
Makes sense.